File: Семестр 1/ДЗ 5/задача_2/list.cpp
#include "list.h"
#include <stdio.h>

List *createList()
{
	List *list = new List;
	list->head = new ListElement;
	list->head->next = NULL;
	list->head->value = 0; 
	return list;
}

Position elemPos(int value, List *list)
{
	ListElement* tmp = list->head;
	while(tmp->value != value)
		tmp=tmp->next;
	return tmp;
}

void addElem(Position p, List *list, int value)
{
	ListElement* tmp = new ListElement;
	tmp->value = value;
	tmp->next = p->next;
	p->next = tmp;
}

void deleteElem(int value, List *list)
{
	ListElement* tmp = list->head;
	while (tmp->next->value != value)
		tmp = tmp->next;
	ListElement *tmp1 = elemPos(value, list);
	tmp->next = tmp1->next;
    delete tmp1;
}

void printList(List *list)
{
	ListElement* tmp = list->head;
	while (tmp->next != NULL)
	{
		printf("%d ", tmp->next->value);
		tmp = tmp->next;
	}
	printf("\n");
}

void deleteList(List *list)
{
	ListElement *tmp = list->head;
	ListElement *tmp1 = tmp->next;
	while (tmp1 != NULL)
	{
		tmp = tmp1;
		tmp1 = tmp1->next;
		delete tmp;
	}
	delete list->head;
	delete list;
}

File: Семестр 2/ДЗ 3/problem_3/problem_3/List.cs
using System;

namespace problem_3
{
    /// <summary>
    /// Singly linked list of pairs of values.
    /// </summary>
    public class List
    {
        /// <summary>
        /// Element of list consists of string and number.
        /// </summary>
        public class ListElement
        {
            public string Str { get; set; }
            public int Count { get; set; }
            public ListElement Next { get; set; }
        }
        private ListElement head;
        public ListElement Head
        {
            get
            {
                return head;
            }
            private set
            {
                head = value;
            }
        }

        /// <summary>
        /// Creates the head of list.
        /// </summary>
        public List()
        {
            Head = new ListElement();
            Head.Next = null;
            Head.Str = "";
            Head.Count = 0;
        }

        /// <summary>
        /// Returns number of elements of the list.
        /// </summary>
        public int SizeOfList()
        {
            int size = 0;
            ListElement tmp = head.Next;
            while (tmp != null)
            {
                ++size;
                tmp = tmp.Next;
            }
            return size;
        }

        /// <summary>
        /// Returns true if there are no elements in the list.
        /// </summary>
        public bool IsEmpty()
        {
            return head.Next == null;
        }

        /// <summary>
        /// Returns the position of the element
        /// </summary>
        /// <param name="value"></param>
        public ListElement ElemPos(string value)
        {
            ListElement tmp = head;
            while (tmp.Str != value)
                tmp = tmp.Next;
            return tmp;
        }

        /// <summary>
        /// Adds new element to the list
        /// </summary>
        /// <param name="pos"></param>
        /// <param name="value"></param>
        public void AddElem(ListElement pos, string value)
        {
	        ListElement tmp = new ListElement();
	        tmp.Str = value;
	        tmp.Next = pos.Next;
	        pos.Next = tmp;
        }

        /// <summary>
        /// Deletes element from the list.
        /// </summary>
        /// <param name="value"></param>
        public void DeleteElem(string value)
        {
            ListElement tmp = head;
            while (tmp.Next.Str != value)
            {
                tmp = tmp.Next;
                if (tmp.Next == null)
                    break;
            }
            if (tmp.Next != null)
            {
                ListElement tmp1 = ElemPos(value);
                tmp.Next = tmp1.Next;
            }
            else
                throw new Exception("Can't find the element in the list.");
        }

        /// <summary>
        /// Prints elements from the first element to the end of list.
        /// </summary>
        public void PrintList()
        {
            ListElement tmp = head;
            while (tmp.Next != null)
            {
                Console.WriteLine(tmp.Next.Str + " " + tmp.Next.Count);
                tmp = tmp.Next;
            }
            Console.WriteLine();
        }
    }
}